plugins/uefi-sbat/fu-uefi-sbat-device.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2024 Richard hughes <Richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-uefi-sbat-device.h"
+#include "fu-uefi-sbat-firmware.h"
+
+struct _FuUefiSbatDevice {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuUefiSbatDevice, fu_uefi_sbat_device, FU_TYPE_DEVICE)
+
+static gboolean
+fu_uefi_sbat_device_probe(FuDevice *device, GError **error)
+{
+	const gchar *vendor;
+	g_autoptr(GHashTable) os_release = NULL;
+	g_autofree gchar *vendor_id = NULL;
+
+	os_release = fwupd_get_os_release(error);
+	if (os_release == NULL)
+		return FALSE;
+	vendor = g_hash_table_lookup(os_release, "ID");
+	if (vendor == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "no os-release ID");
+		return FALSE;
+	}
+	vendor_id = g_strdup_printf("OS:%s", vendor);
+	fu_device_add_vendor_id(device, vendor_id);
+
+	/* try to lookup /etc/os-release ID key */
+	fu_device_add_instance_str(device, "VENDOR", vendor);
```

Comment:
It's not really "Vendor" though.  The ID key refers to a distro.

https://www.freedesktop.org/software/systemd/man/latest/os-release.html#ID=

Reply:
Yes, good catch. I've changed it to be `DISTRO`.

Reply:
Actually, maybe OS makes more sense in this context. I'll use that instead.